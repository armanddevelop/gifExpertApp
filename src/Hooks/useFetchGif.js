import { useState, useEffect } from "react";
import { getGifts } from "../utils/Apis";

export const useFetchGift = (category) => {
  const [stateData, setStateData] = useState({
    data: [],
    loading: true,
  });
  useEffect(() => {
    getGifts(category).then((data) => {
      setStateData({
        data,
        loading: false,
      });
    });
  }, [category]);
  return stateData;
};
