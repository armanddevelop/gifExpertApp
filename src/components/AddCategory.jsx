import { useState } from "react";
import PropTypes from "prop-types";
export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");
  const handleChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    inputValue.trim().length > 3 &&
      setCategories((category) => [inputValue, ...category]);
    setInputValue("");
  };
  return (
    <form onSubmit={handleSubmit} noValidate>
      <input
        placeholder="Search for a Gif"
        type="text"
        value={inputValue}
        onChange={(e) => handleChange(e)}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
