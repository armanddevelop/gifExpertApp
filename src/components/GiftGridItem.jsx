export const GiftGridItem = ({ img }) => {
  const { title, url, username } = img;
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={username} />
      <p>{title === "" ? "no title" : title}</p>
    </div>
  );
};
