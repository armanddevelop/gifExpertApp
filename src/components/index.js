export { GiftGridItem } from "./GiftGridItem";
export { GiftGrid } from "./GiftGrid";
export { AddCategory } from "./AddCategory";
