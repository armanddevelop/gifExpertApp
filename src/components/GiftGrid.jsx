import { useFetchGift } from "../Hooks/useFetchGif";
import { GiftGridItem } from ".";
import PropTypes from "prop-types";
export const GiftGrid = ({ category }) => {
  const { data: images, loading } = useFetchGift(category);
  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="animate__animated animate__flash">Cargando.....</p>
      )}
      <div className="card-grid">
        {images.map((image) => (
          <GiftGridItem key={image.id} img={image} />
        ))}
      </div>
    </>
  );
};
GiftGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
