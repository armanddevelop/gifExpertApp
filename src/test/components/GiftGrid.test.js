import { GiftGrid } from "../../components";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { useFetchGift } from "../../Hooks/useFetchGif";
jest.mock("../../Hooks/useFetchGif");

describe("Test to GiftGrid Component", () => {
  const category = "one punch";
  it("snapshot for the GiftGrid component", () => {
    useFetchGift.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(true);
  });
  it("should show items when images are load useFetchGift", () => {
    const gifs = [
      {
        id: 123,
        username: "creting",
        title: "euche",
        url: "https://www.hola.com",
      },
    ];
    useFetchGift.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GiftGridItem").length).toBe(gifs.length);
  });
});
