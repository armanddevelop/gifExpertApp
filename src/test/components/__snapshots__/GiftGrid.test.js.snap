// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GiftGrid Component should show items when images are load useFetchGift 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <div
    className="card-grid"
  >
    <GiftGridItem
      img={
        Object {
          "id": 123,
          "title": "euche",
          "url": "https://www.hola.com",
          "username": "creting",
        }
      }
      key="123"
    />
  </div>
</Fragment>
`;

exports[`Test to GiftGrid Component snapshot for the GiftGrid component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    one punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando.....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
